proto/zitadel/user/v2/user_service.proto

```diff
@@ -1232,6 +1234,55 @@ service UserService {
     };
   }
 
+  // ListMetadata returns all matching metadata entries for a user.
+  // You can filter by key only. Filtering by value is not supported yet.
+  // If no filter is specified, all metadata entries are returned.
+  //
+  // Required permissions:
+  //   - 'user.read'
+  //   - no permissions required for listing own metadata
+  rpc ListMetadata (ListMetadataRequest) returns (ListMetadataResponse) {}
+
+  // GetMetadata returns a single metadata entry of the user identified by the key.
+  //
+  // Required permissions:
+  //   - 'user.read'
+  //   - no permissions required for getting own metadata
+  rpc GetMetadata (GetMetadataRequest) returns (GetMetadataResponse) {}
+
+  // AddMetadata adds a new metadata entry to the user.
+  // If the key already exists, an error is returned.
+  //
+  // Required permissions:
+  //   - 'user.write'
+  rpc AddMetadata (AddMetadataRequest) returns (AddMetadataResponse) {}
+
+  // UpdateMetadata updates an existing metadata entry of the user.
+  // If the key does not exist, an error is returned.
+  //
+  // Required permissions:
+  //   - 'user.write'
+  rpc UpdateMetadata (UpdateMetadataRequest) returns (UpdateMetadataResponse) {}
+
+  // Set Metadata adds or updates a metadata entry of the user identified by the key.
+  // If the key does not exist, a new entry is created.
+  // If the key already exists, the value is updated.
+  // In case a key from an existing entry needs to be updated,
+  // the existing entry must be removed and a new one created.
+  //
+  // Required permissions:
+  //   - 'user.write'
+  rpc SetMetadata (SetMetadataRequest) returns (SetMetadataResponse) {}
```

Comment:
#9165 specifies `Add` and `Update`, but IMO we should just go with `Set`

Reply:
+1 for set, replacing add and update